gradients-only/bin/main.ml

```diff
@@ -8,11 +8,9 @@ let tick t screen _prev _inputs =
     let yf = float_of_int y /. float_of_int height in
 
     let shift = (sin (float_of_int t /. 50.0) +. 1.0) /. 2.0 in
-    let r = int_of_float (255. *. xf *. shift) in
-    let g = int_of_float (255. *. yf *. shift) in
-    let b = int_of_float (255. *. (1.0 -. shift)) in
-
-    (r lsl 16) lor (g lsl 8) lor b
+    let gradient_value = int_of_float (255. *. (xf *. 0.5 +. yf *. 0.5 +. shift *. 0.5)) in
+    
+    gradient_value mod 256  
```

Comment:
Excellent! You're now using all the variables (x and y position, and time) for just the one return value, and you're also explicitly clipping it to the palette size!

If you wanted to make the code more flexible, you can also look up the size of the palette in your tick function, so that if you ever changed the palette size, this code wouldn't need updating. You can see an example of that in the day2 example:

https://github.com/claudiusFX/claudius-examples/blob/main/day2/bin/main.ml#L4